core/src/main/java/com/novoda/espresso/ViewTestRule.java

```diff
@@ -1,50 +1,73 @@
 package com.novoda.espresso;
 
+import android.app.Activity;
 import android.app.Instrumentation;
-import android.content.Intent;
 import android.support.annotation.LayoutRes;
 import android.support.test.InstrumentationRegistry;
 import android.support.test.rule.ActivityTestRule;
 import android.view.View;
+import android.view.ViewGroup;
 
-public class ViewTestRule<T extends View> extends ActivityTestRule<ViewActivity> {
+import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;
+
+public class ViewTestRule<T extends View> extends ActivityTestRule<EmptyActivity> {
 
     private final Instrumentation instrumentation;
+    private final ViewCreator<T> viewCreator;
+
+    private T view;
 
-    @LayoutRes
-    private final int id;
+    public ViewTestRule(@LayoutRes int layoutId) {
```

Comment:
personally I would prefer these were factory methods with the ctor hidden

```
ViewTestRule.from(R.layout.foobar / viewCreator);
```

If someone wants to modify the `Instrumentation` instance they wouldn't be able to use the rule lifecycle anyways

Reply:
can do

Reply:
commit 8bb51fe4dd5e3f4590c6754669b5ede7484d5091
Author: Ataul Munim <hello@ataulm.com>
Date:   Wed Nov 8 13:11:52 2017 +0000

    Reduce scope of test rule ctors

just reduced the scope of the ctor that takes Instrumentation, didn't pull out named ctors yet - can defer that decision for another time.

the reason is I can't think what `ViewTestRule.from(@LayoutRes int)` gives over `new ViewTestRule(@LayoutRes int)`.

Originally had `ViewTestRule.withViewFromXml(@LayoutRes int)` but this is the only named one I could think of. If there's more, we can def pull out to help understanding.